src/store.tsx

```diff
@@ -0,0 +1,117 @@
+import { useReducer, useEffect, createContext, useContext, useCallback, useMemo, useState } from "react";
+
+export interface Movie {
+  id: string;
+  name: string;
+  genres: string[];
+  year: number;
+  poster: string;
+}
+
+function useMoviesSource(): {
+  movies: Movie[];
+  filterBy: string;
+  setFilterBy: (filterBy: string) => void;
+  sortBy: string;
+  setSortBy: (sortBy: string) => void;
+  selectedMovie: Movie | undefined;
+  setSelectedMovieId: (movieId: string) => void;
+} {
+  type MoviesState = {
+    movies: Movie[];
+    filterBy: string;
+    sortBy: string;
+  };
+
+  type MoviesAction =
+    | { type: "setMovies"; payload: Movie[] }
+    | { type: "setFilterBy"; payload: string }
+    | { type: "setSortBy"; payload: string };
+
+  const [{ movies, filterBy, sortBy }, dispatch] = useReducer(
+    (state: MoviesState, action: MoviesAction) => {
+      switch (action.type) {
+        case "setMovies":
+          return { ...state, movies: action.payload };
+        case "setFilterBy":
+          return { ...state, filterBy: action.payload };
+        case "setSortBy":
+          return { ...state, sortBy: action.payload };
+      }
+    },
+    {
+      movies: [],
+      filterBy: "",
+      sortBy: "",
+    }
+  );
```

Comment:
I like that you've used `useReducer`. Let's make it a bit more readable. 
`useReducer` expects 2 arguments: reducer and default store. Let's move them to a separare named variables. 
```suggestion
  const [{ movies, filterBy, sortBy }, dispatch] = useReducer(
   reducer,
   defaultStore
  );
```

Reply:
Not Resolved 👀